React/tic-tac-toe/src/redux/login/actions.js

```diff
@@ -1,16 +1,45 @@
-export const actions = {
-  GET_LOGGED_USER: 'GET_LOGGED_USER',
-  SET_LOGGED_USER: 'SET_LOGGED_USER'
-};
+import { stringArrayToObject } from '../../utils/stringUtils';
+import { login as loginService } from '../../services/LoginService';
+
+export const actions = stringArrayToObject(
+  [
+    'LOGIN_REQUEST',
+    'LOGIN_SUCCESS',
+    'LOGIN_FAILURE'
+  ],
+  '@@LOGIN'
+);
 
 const actionCreators = {
-  getLoggedUser: () => ({
-    type: actions.GET_LOGGED_USER
-  }),
-  setLoggedUser: status => ({
-    type: actions.SET_LOGGED_USER,
-    logged: status
-  })
+  login: (user) => async dispatch => {
+    dispatch({ type: actions.LOGIN_REQUEST });
+    const response = await loginService({ email: user.email });
+    if (response.ok) {
+      if (response.data.length > 0) {
+        if (response.data[0].password === user.password) {
+          dispatch({
```

Comment:
you should have only one dispatch per action. The right thing to do is to call a private action here. Like this:
```
 getVehicleInformation(vehicleId) {
    return async dispatch => {
      dispatch({ type: actions.GET_VEHICLE_INFORMATION });
      try {
        const response = await getVehicleProfile(vehicleId);
        if (response.ok) {
          dispatch(privateActionCreators.getVehicleInformationSuccess(response.data));
        } else {
          throw new Error(response.problem);
        }
      } catch (e) {
        dispatch(privateActionCreators.getVehicleInformationFailure(e.message));
      }
    };
  }
```

Then you will have the private action similar to:
```
const privateActionCreators = {
  getPositionSuccess(data, target) {
    return {
      type: actions.GET_POSITION_SUCCESS,
      payload: { data },
      target
    };
  },
  getPositionFailure(err) {
    return {
      type: actions.GET_POSITION_FAILURE,
      payload: { err }
    };
  },
  getRuleStationsSuccess(data) {
    return { type: actions.GET_RULE_STATIONS_SUCCESS, payload: { data } };
  },
  getRuleStationsFailure(err) {
    return { type: actions.GET_RULE_STATIONS_FAILURE, payload: { err } };
  },
  getVehicleInformationSuccess(data) {
    return { type: actions.GET_VEHICLE_INFORMATION_SUCCESS, payload: { data } };
  },
  getVehicleInformationFailure(err) {
    return { type: actions.GET_VEHICLE_INFORMATION_FAILURE, payload: { err } };
  }
};
```